Source/DafnyCore/AST/MemberDecls.cs

```diff
@@ -466,6 +467,34 @@ public class Method : MemberDecl, TypeParameter.ParentType, IMethodCodeContext {
   public override bool IsOverrideThatAddsBody => base.IsOverrideThatAddsBody && Body != null;
   private static BlockStmt emptyBody = new BlockStmt(Token.NoToken, Token.NoToken, new List<Statement>());
 
+  public override IEnumerable<AssumptionDescription> Assumptions {
+    get {
+      foreach (var a in base.Assumptions) {
+        yield return a;
+      }
+
+      if (Children.Any(n => n is AssumeStmt)) {
+        yield return AssumptionDescription.AssumeInBody;
+      }
+
+      if (Body is null && Ens.Count > 0) {
+        yield return AssumptionDescription.NoBody(IsGhost);
+      }
+
+      if (Attributes.Contains(Attributes, "extern") && Ens.Count > 0) {
```

Comment:
What about implicit clauses coming from subtypes in the arguments? What about the modifies clause? Shouldn't anything with `{:extern}` on it show up in the report since there's always an (implicit) modifies clause. Explicitly specifying `modifies *` is not allowed by the language.

Reply:
Yeah, this should include any potential subtypes on the return type, as well. For functions, I think anything that doesn't have a postcondition (including a constraint on the return type) is safe. But I agree that we should always include methods. Thanks for pointing that out.

Reply:
Right, if there is a subtype in the return value, it's like having an ensures.

Functions could be implemented by methods that modify the state, so it's never possible to be sure of anything. But I guess we can leave it out for later.